### ISSUE TITLE
Ignition fills console buffer with its config when failing

### ISSUE THREAD

**Issue by PhrozenByte** [coreos/ignition#1288]
**Describe the bug**
Ignition will print its full config when failing on first boot. For larger configs this will fill the whole console buffer (i.e. hitting the max scrollback size), making it impossible to see the actual errors causing Ignition to fail. This makes debugging virtually impossible.

**Reproduction steps**
1. Create a large Ignition config that will fail on first boot for whatever reason (e.g. trying to create an already existing file without setting `overwrite: true`).
2. Run `coreos-installer` and reboot the system.
3. See how Ignition fills the whole console buffer by printing its config.

**Expected behavior**
Ignition shouldn't print its full config when failing. This often is an useless information anyway, because one can usually retrieve the same config by other means anyway. Optionally there could be a kernel option to enable this manually.

**Actual behavior**
Ignition prints its full config filling up the console buffer.

**System details**
 - Bare Metal
 - FCOS 35.20211029.3.0

**Ignition config**
Not relevant.

**Additional information**
None.

**Comment by bgilbert** [comment]
The main reason we print the full config is to detect the case where the machine is using an unintended config.  You're right, though, that we might not be making the right tradeoff.

A kernel option doesn't really help because in some situations there isn't a good way to override kernel options.

**Comment by PhrozenByte** [comment]
I see. I guess we could achieve the same thing by calculating a shortened `sha1` hash of the config and printing this hash instead. `coreos-installer` and Butane could both do the same, so one can compare the hashes.

**Comment by jlebon** [comment]
Ignition today does print the hash of all the configs it fetches, but they're only in the journal, not the console. We could tweak that so it goes to both. But even then, hashes aren't super friendly. Maybe simplest is to keep the config that failed to parse somewhere and just print instructions for how to look at it?

**Comment by PhrozenByte** [comment]
I agree. Printing hashes *and* writing the config somewhere to inspect later (possibly the journal) is probably the best way to go. I agree that comparing hashes isn't super friendly either. But comparing two (possibly huge) JSON dumps is harder. If one just wants to verify whether the expected config was used, one is best with comparing hashes - and if Ignition uses crypto hashes, people likely recognize different hashes at first glance anyway.

**Comment by PhrozenByte** [comment]
I'm usually not bumping tickets, but this is a super annoying issue. It often leaves me with just guessing whatever might have caused the error. It's basically unusable with larger ignition configs.

@jlebon: You said that the error is also written to the journal, however, I'm not sure what journal you mean? If Ignition fails on first boot, the journal isn't actually written to disk, is it? Because I couldn't find the error using `journalctl`.

**Comment by bgilbert** [comment]
The journal isn't written to disk, but you can press Enter for an emergency shell prompt and type `journalctl -t ignition` to page through the complete Ignition logs.
